packages/core/src/base/staticClient.ts

```diff
@@ -57,6 +57,19 @@ async function runFetch(
     ? _.clone(options.client.clientConfig.headers)
     : {};
 
+  // make-fetch-happen sets connection header to keep-alive by default which
+  // keeps node running unless it is explicitly killed.
+  // If the user wants to keep the connection alive they can set the Connection
+  // header to 'keep-clive' and we'll respect it. Otherwise, we set it to false.
+  //
+  // explicitly make connection header lowercase because otherwise 
+  // make-fetch-happen makes it lowercase later, creates an array by merging the 
+  // values, and we end up with two values.
+  fetchOptions.headers.connection = fetchOptions.headers.connection
+    ? fetchOptions.headers.connection : fetchOptions.headers.Connection ?
+      fetchOptions.headers.Connection : "close";
+  delete fetchOptions.headers.Connection;
```

Comment:
We should add a couple exported functions to this file to help with setting headers that are pretty much the find/get/set from https://github.com/npm/minipass-fetch/blob/master/lib/headers.js